BuildProject/App.tsx

```diff
@@ -147,6 +147,28 @@ const App: () => Node = () => {
     recordTestPassed('setAncestorIds');
   }
 
+  try {
+    useImpressionTracker(
+      (viewToken) => ({
+        contentId: 'foo',
+        insertionId: 'bar',
+        name: 'batman'
+      }),
+      'TestCollectionViewName',
+      ImpressionSourceType.Delivery
+    );
+    useViewTracker(useRef());
+    useEffect(() => {
+      setText('Passed: useImpressionTracker\n' +
+              'Passed: useViewTracker\n' +
+              'All hooks passed');
+    }, []);
+  } catch (err) {
+    useEffect(() => {
```

Comment:
side-comment - Interesting.  I don't know React Native.  In normal React, we shouldn't change the number of React Hooks in a call based on conditions and cases that happen in it.  I haven't seen the pattern with catching.  I think they recommend error boundaries for these types of exceptions.

Reply:
The constraint with hooks is identical in React Native. Here, I'm only checking the exceptional case where one of `useImpressionTracker` or `useViewTracker` was broken by a change in some PR. If one of those crashed, the test would fail anyways.

Maybe it's better to not catch any errors and let them propagate, so that causes the test to fail. I was hoping that in this case it would at least allow the test to report the error that caused the crash though. Let me think about how to get the error message to appear in the failing test without violating the hook invariant.